src/main/java/uk/ac/cam/cl/dtg/isaac/api/PagesFacade.java

```diff
@@ -388,40 +381,75 @@ public final Response getQuestionList(@Context final Request request,
+            try {
+                ResultsWrapper<ContentDTO> c;
+                c = contentManager.searchForContent(
+                        validatedSearchString,
+                        fieldsToMatch.get(ID_FIELDNAME),
+                        fieldsToMatch.get(TAGS_FIELDNAME),
+                        fieldsToMatch.get(SUBJECTS_FIELDNAME),
+                        fieldsToMatch.get(FIELDS_FIELDNAME),
+                        fieldsToMatch.get(TOPICS_FIELDNAME),
+                        fieldsToMatch.get(BOOKS_FIELDNAME),
+                        fieldsToMatch.get(LEVEL_FIELDNAME),
+                        fieldsToMatch.get(STAGE_FIELDNAME),
+                        fieldsToMatch.get(DIFFICULTY_FIELDNAME),
+                        fieldsToMatch.get(EXAM_BOARD_FIELDNAME),
+                        fieldsToMatch.getOrDefault(TYPE_FIELDNAME, Set.of(QUESTION_TYPE)),
+                        nextSearchStartIndex,
+                        newLimit,
+                        showNoFilterContent
+                );
 
-        } catch (ContentManagerException e1) {
-            SegueErrorResponse error = new SegueErrorResponse(Status.NOT_FOUND,
-                    "Error locating the content requested", e1);
-            log.error(error.getErrorMessage(), e1);
-            return error.toResponse();
+                summarizedResults = this.extractContentSummaryFromList(c.getResults());
+                resultsReturnedByThisSearch = summarizedResults.size();
+                nextSearchStartIndex += resultsReturnedByThisSearch; // add the total number of returned results, including any we will filter out
+
+                if (user instanceof RegisteredUserDTO) {
+                    RegisteredUserDTO registeredUser = (RegisteredUserDTO) user;
+                    List<String> questionPageIds = summarizedResults.stream().map(ContentSummaryDTO::getId).collect(Collectors.toList());
+
+                    Map<String, Map<String, List<LightweightQuestionValidationResponse>>> questionAttempts = questionManager
+                            .getMatchingLightweightQuestionAttempts(Collections.singletonList(registeredUser), questionPageIds)
+                            .getOrDefault(registeredUser.getId(), Collections.emptyMap());
+
+                    for (ContentSummaryDTO result : summarizedResults) {
+                        augmentContentSummaryWithAttemptInformation(result, questionAttempts);
+                    }
+
+                    if (hideCompleted) {
+                        summarizedResults = summarizedResults.stream()
+                                .filter(q -> q.getCorrect() == null || !q.getCorrect())
+                                .collect(Collectors.toList());
+                    }
+                }
```

Comment:
This is the sort of logic that is supposed to live in Manager classes, not Facade classes. But it's currently not clear to me which manager this is best suited to be in, and there was already a lot of logic in this method beforehand :/